src/android/com/adobe/phonegap/push/GCMIntentService.java

```diff
@@ -229,6 +229,21 @@ private Bundle normalizeExtras(Context context, Bundle extras) {
                         Log.e(LOG_TAG, "normalizeExtras: JSON exception");
                     }
                 }
+            } else if (key.equals(APPCELERATOR_PAYLOAD)) {
+                try {
+                    String json = extras.getString(key);
+                    JSONObject data = new JSONObject((String) json);
```

Comment:
Might as well combine lines 234 and 235. Also do you need the explicit `String` cast?